### PR TITLE
Fix linux tray settings

### PR DESCRIPTION
**Pull request by itsmartin**
- [x] Complete [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
- [x] Execute `npm run prettify` to format codes
- [x] Write about environment which you tested
  - Operating system: Linux, xubuntu 16.04.1 LTS (xfce 4.12)

This is a small PR to fix a couple of minor issues with the settings page on Linux.

- The option "Leave app running in notification area when the window is closed" was never enabled, even after restarting the app, because the check in `main.js` that sets `trayWasVisible` was not running.
- Additionally, one of the settings was wrongly named, referring to the "menu bar" when it should have said "notification area", so I've updated this too.

### CODE REVIEW

**Comment by jasonblais** [comment]
Awesome, thanks @itsmartin!!

Looks like the build is failing, @yuya-oc any hints to help fix it?

File: src/browser/settings.jsx [hunk 1]
```diff
@@ -168,7 +168,7 @@ var SettingsPage = React.createClass({
                      onChange={ this.handleChangeHideMenuBar } />);
     }
     if (process.platform === 'darwin' || process.platform === 'linux') {
-      options.push(<Input key="inputShowTrayIcon" id="inputShowTrayIcon" ref="showTrayIcon" type="checkbox" label="Show icon on menu bar (Need to restart the application)" checked={ this.state.showTrayIcon }
+      options.push(<Input key="inputShowTrayIcon" id="inputShowTrayIcon" ref="showTrayIcon" type="checkbox" label="Show icon in notification area (Need to restart the application)" checked={ this.state.showTrayIcon }
```
Comment (yuya-oc) [inline src/browser/settings.jsx]: "menu bar" is the correct name in macOS. So you should prepare two cases here.

**Comment by yuya-oc** [comment]
Error is only for code formatter, so we can test the artifacts. https://circleci.com/gh/mattermost/desktop/709#artifacts

**Comment by jasonblais** [comment]
Would anyone in the Linux community like to test?

Looking for two reviewers, you can download the builds below:
 - [../ia32.tar.gz](https://709-43964563-gh.circle-artifacts.com/0/tmp/circle-artifacts.UIbHlhQ/mattermost-desktop-linux-ia32.tar.gz)
 - [../x64.tar.gz](https://709-43964563-gh.circle-artifacts.com/0/tmp/circle-artifacts.UIbHlhQ/mattermost-desktop-linux-x64.tar.gz)
 - [../i386.deb](https://709-43964563-gh.circle-artifacts.com/0/tmp/circle-artifacts.UIbHlhQ/mattermost-3.4.0-ia32.deb)
 - [../amd64.deb](https://709-43964563-gh.circle-artifacts.com/0/tmp/circle-artifacts.UIbHlhQ/mattermost-3.4.0.deb)

**Comment by ldziedziul** [comment]
I've tested x64.tar.gz. The label was changed to "Show icon in notification area" and "Leave app running in notification area when the window is closed" is enabled when "Show icon in notification area" is checked. "Leave app..." doesn't change anything. No matter if it's checked or not application stays open after closing app window.

File: src/browser/settings.jsx (hunk 1 above)
Comment (itsmartin) [inline src/browser/settings.jsx]: Confusing! Updated the commit.

**Comment by itsmartin** [comment]
@ldziedziul That sounds like it's unrelated to this change. Just to check, did you fully quit and restart the application?

**Comment by ldziedziul** [comment]
@itsmartin yes, I've killed it and run it again.

**Comment by jasonblais** [comment]
Thanks @ldziedziul for testing, much appreciated!


Just to clarify: If `Show icon in notification area` is enabled, closing the application window (from "X" on the top right corner of the app) keeps Mattermost in the notification area and therefore covers the `Leave app running in notification area when the window is closed` setting automatically?

 Is this behaviour consistent with the v3.4 release build?

**Comment by jasonblais** [comment]
Would you @yuya-oc like to test it out and see how it works in your environment?

**Comment by yuya-oc** [comment]
Tested on Ubuntu 14.04 Unity. I think the feature works well.
If "Leave app..." is enabled, the taskbar icon disappears when closing the window. Then, it's restored when I have selected a team in the tray icon menu.

File: src/browser/settings.jsx [hunk 2]
```diff
@@ -168,7 +168,7 @@ var SettingsPage = React.createClass({
                      onChange={ this.handleChangeHideMenuBar } />);
     }
     if (process.platform === 'darwin' || process.platform === 'linux') {
-      options.push(<Input key="inputShowTrayIcon" id="inputShowTrayIcon" ref="showTrayIcon" type="checkbox" label="Show icon on menu bar (Need to restart the application)" checked={ this.state.showTrayIcon }
+      options.push(<Input key="inputShowTrayIcon" id="inputShowTrayIcon" ref="showTrayIcon" type="checkbox" label={ process.platform === 'darwin' ? "Show icon on menu bar (need to restart the application)" : "Show icon in notification area (need to restart the application)" } checked={ this.state.showTrayIcon }
```
Comment (yuya-oc) [inline src/browser/settings.jsx]: Probably this should be like:

```jsx
options.push(<Input key="inputShowTrayIcon" id="inputShowTrayIcon" ref="showTrayIcon" type="checkbox" label={ process.platform === 'darwin' ? "Show icon on menu bar (need to restart the application)" : "Show icon in notification area (need to restart the application)" } checked={ this.state.showTrayIcon } onChange={ this.handleChangeShowTrayIcon }
             />);
```

**Comment by jasonblais** [comment]
Thanks @yuya-oc! 

@itsmartin would you like to check the note above? Thanks again for submitting the pull request, much appreciated

File: src/browser/settings.jsx (hunk 2 above)
Comment (jasonblais) [inline src/browser/settings.jsx]: @itsmartin Many thanks again for your pull request, highly appreciated! 

Would you like to take a look at the note above? We'll be pretty close to merging the PR after the change.

**Comment by tfgm-bud** [comment]
@jasonblais I tested the x64.tar version on Linux Mint 18 Sarah.  Works well...

**Comment by tfgm-bud** [comment]
@jasonblais oh, I should mention that when I run from the command line I see an "error" but it seem to run fine:

```bash
$ ./Mattermost 
{ Error: ENOENT: no such file or directory, open '/home/bud/.config/Mattermost/certificate.json'
    at Error (native)
    at Object.fs.openSync (fs.js:634:18)
    at Object.module.(anonymous function) [as openSync] (ELECTRON_ASAR.js:158:20)
    at Object.fs.readFileSync (fs.js:502:33)
    at Object.fs.readFileSync (ELECTRON_ASAR.js:426:29)
    at new CertificateStore (/storage/bud/Downloads/mattermost-desktop-linux-x64/resources/app/main.js:2847:32)
    at Object.load (/storage/bud/Downloads/mattermost-desktop-linux-x64/resources/app/main.js:2877:13)
    at Object.<anonymous> (/storage/bud/Downloads/mattermost-desktop-linux-x64/resources/app/main.js:98:49)
    at __webpack_require__ (/storage/bud/Downloads/mattermost-desktop-linux-x64/resources/app/main.js:20:30)
    at __webpack_require__.app.app (/storage/bud/Downloads/mattermost-desktop-linux-x64/resources/app/main.js:40:18)
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/home/bud/.config/Mattermost/certificate.json' }
Created new window in existing browser session.
Created new window in existing browser session.
```

**Comment by yuya-oc** [comment]
@tfgm-bud No problem. But it's confusing, so probably it should be suppressed.

**Comment by yuya-oc** [comment]
The reason for blocking this PR is just coding style. It's important, but this will be rewritten by my coming commits. So I specially merge this to avoid conflicting.

**Pull request merged**